lib/oli_web/live/history/revision_history.ex

```diff
@@ -360,6 +361,64 @@ defmodule OliWeb.RevisionHistory do
     end
   end
 
+  def handle_event("reset-monaco", _params, socket) do
+    {:noreply, assign(socket, details_modal_assigns: nil)}
+  end
+
+  def handle_event("edit-attribute", %{"attr-key" => attr_key}, socket) do
+    socket =
+      assign(socket,
+        details_modal_assigns: %{
+          title: attr_key,
+          key: String.to_existing_atom(attr_key)
+        }
+      )
+
+    # we want to ensure monaco editor is reseted correctly
+    # before assigning new values and showing the modal
+    :timer.sleep(200)
```

Comment:
The `reset-monaco` event is dispatched before this event so why is this necessary?

Reply:
I had the same thought as you and, while developing the feature and testing it locally, sometimes I opened the modal for a new edit and got the values of the previous edited field. 
To be honest, I really do not like having to add a `timer.sleep`. But from the user's perspective, this delay is unnoticeable, the modal seems to open immediately. And having to close the modal and reopen it when you get the wrong values is really annoying.